aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,795 @@
+                    not in [
+                        DimensionNames.SpatialY,
+                        DimensionNames.SpatialX,
+                        CZI_SAMPLES_DIM_CHAR,
+                    ]
+                    and dim in dims_shape.keys()
+                ):
+                    # Handle slices
+                    if index_op is None:
+                        # Store the dim for later to inform to use the np index
+                        use_selected_or_np_map[dim] = None
+                        if dim == DimensionNames.MosaicTile:
+                            retrieve_shape.append(
+                                dims_shape[DimensionNames.MosaicTile][1]
+                            )
+                        elif dim == DimensionNames.Time:
+                            retrieve_shape.append(dims_shape[DimensionNames.Time][1])
+                        elif dim == DimensionNames.Channel:
+                            retrieve_shape.append(dims_shape[DimensionNames.Channel][1])
+                        elif dim == DimensionNames.SpatialZ:
+                            retrieve_shape.append(
+                                dims_shape[DimensionNames.SpatialZ][1]
+                            )
+
+                    # Handle non-chunk dimensions (specific indices / ints)
+                    else:
+                        # Store the dim for later to inform to use the provided index
+                        use_selected_or_np_map[dim] = index_op
+                        retrieve_shape.append(1)
+
+            # Create list of planes that we will add each plane to, later we reshape
+            # Create empty arr with the desired shape to enumerate over the np index
+            planes: List[np.ndarray] = []
+            np_array_for_indices = np.empty(tuple(retrieve_shape), dtype=object)
+            for np_index, _ in np.ndenumerate(np_array_for_indices):
+                # Get each plane's index selection operations
+                # If the dimension is None, use the enumerated np index
+                # If the dimension is not None, use the provided value
+                plane_indices: Dict[str, int] = {"S": scene}
+
+                for dim in [
+                    DimensionNames.MosaicTile,
+                    DimensionNames.Time,
+                    DimensionNames.Channel,
+                    DimensionNames.SpatialZ,
+                ]:
+                    indexes = CziReader._dim_helper(
+                        dim,
+                        use_selected_or_np_map,
+                        np_index,
+                        retrieve_dims,
+                    )
+                    if indexes is not None:
+                        plane_indices[dim] = indexes
+
+                # Append the retrieved plane as a numpy array
+                plane, scene_dims = czi.read_image(**plane_indices)
+                scene_dims_dict = CziReader._dims_list_to_dict(scene_dims)
+                planes.append(plane)
```

Comment:
Just now realizing... Are you pulling single planes at a time? Can't aicspylibczi pull a whole chunk at a time? I think this is one of those "copied from `LifReader`.... `readlif` can only pull a single plane at a time which is why I had to do it that way but it would be faster IO if you can pull a whole chunk at a time.

Reply:
See TiffReader for this behavior I believe.

Reply:
We can also make that optimization later.

Reply:
I worked from the LifReader so yeah I fell into that trap. I'll rework it now. Thanks for catching this.